lib/server/services/lhcPeriod/LhcPeriodStatisticsService.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { repositories: { LhcPeriodRepository } } = require('../../../database');
+const { dataSource } = require('../../../database/DataSource.js');
+const { ApiConfig } = require('../../../config');
+const { lhcPeriodAdapter } = require('../../../database/adapters');
+
+/**
+ * @typedef LhcPeriodIdentifier object to uniquely identify a lhc period
+ * @property {string} [name] the lhc period name
+ * @property {number} [id] the id of lhc period
+ */
+
+/**
+ * LhcPeriodStatisticsService
+ */
+class LhcPeriodStatisticsService {
+    /**
+     * Find a lhc period model by its name or id
+     * @param {LhcPeriodIdentifier} identifier the criteria to find run type
+     * @return {Promise<SequelizeLhcPeriod[]>} the run type found or null
+     */
+    async getByIdentifier({ id, name }) {
+        const queryBuilder = dataSource.createQueryBuilder();
+        if (id) {
+            queryBuilder.where('id').is(id);
+        }
+        if (name) {
+            queryBuilder.where('name').is(name);
+        }
+        queryBuilder.include({ association: 'lhcPeriodStatistics' });
+
+        const lhcPeriod = await LhcPeriodRepository.findOne(queryBuilder);
+        return lhcPeriod ? lhcPeriodAdapter.toEntity(lhcPeriod) : null;
+    }
+
+    /**
+     * Get all lhc periods with statistics
+     * @param {Object} [options.filter] definition of filtering
+     * @param {Number} [options.page.offset] definition of pagination - offset
+     * @param {Number} [options.page.limit] definition of pagination - limit
+     * @param {Object<String, String>[]} [options.sort] definition of sorting - array of objects like { [FIELD_NAME]: 'DESC' or 'ASC' }
+     * @returns {CountedData} result
+     */
+    async getAllForPhysicsRuns({ filter, page = {}, sort = { name: 'desc' } } = {}) {
+        const queryBuilder = dataSource.createQueryBuilder();
+
+        for (const property in sort) {
+            queryBuilder.orderBy(property, sort[property]);
+        }
+        const { limit = ApiConfig.pagination.limit, offset = 0 } = page;
+        queryBuilder.limit(limit);
+        queryBuilder.offset(offset);
+
+        if (filter) {
+            const { ids, names } = filter;
+            if (ids) {
+                queryBuilder.where('id').oneOf(...Array.isArray(ids) ? ids : [ids]);
```

Comment:
I see no reason at this point to allow id to not be an array. Same for name

Reply:
with this you may put query params  like `?names=LHC22a` instead of `?names[]=LHC22a`. To be honest I don't care

Reply:
If you mention `query params` stuff it means that it should not be in services

Reply:
I mentioned it, because, finally, scope with `'query params'` utilize scope of the service, but if its matters I can rephrase my previous comment to <b>with this you may call the service `#getAllForPhysicsRuns({filter: {names: 'LHC22a'}})` instead of `#getAllForPhysicsRuns({filter: {names: ['LHC22a']}})`  </b>.
And now ask straight out, constraint it to accept only arrays?

Reply:
so remove handling non-array value, ok

Reply:
done